nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/ApplicationLifecycle.java

```diff
@@ -548,6 +548,7 @@ public void actOn(Logger logger) {
             // first, then register the real one
             appRegistry.remove(appName);
             appInfo.setIsJavaEEApp(sortedEngineInfos);
+            appInfo.setTypes(context.getTransientAppMetaData(Types.class.getName(), Types.class));
```

Comment:
This should be lazily done on demand. If OpenAPI is requesting the types we still could compute this data from the sources used here or am I missing something?

Reply:
`org.glassfish.hk2.classmodel.reflect.Types` data was always created and loaded during application deployment and removed after deployments completed. 

Now `Types` info is stored in the `ApplicationInfo` for further processing by other modules like OpneAPI.